test/deprecated.jl

```diff
@@ -22,3 +22,9 @@ end
     E = ChainRulesCore.NotImplementedException
     @test_throws E extern(ni)
 end
+
+
+@testset "Deprecated: calling thunks should call inner function" begin
+    @test_deprecated (@thunk(3))() == 3
+    @test_deprecated (@thunk(@thunk(3)))() isa Thunk
```

Comment:
Should we also test `InplaceableThunk`?

Reply:
> Should we also test InplaceableThunk?

Well, we didn't before, but I can add it.